var/spack/repos/builtin/packages/netcdf/package.py

```diff
@@ -11,13 +12,15 @@ class Netcdf(Package):
     version('4.4.0', 'f01cb26a0126dd9a6224e76472d25f6c')
     version('4.3.3', '5fbd0e108a54bd82cb5702a73f56d2ae')
 
+    variant('mpi', default=True, description='Enables MPI parallelism')
     variant('fortran', default=False, description="Download and install NetCDF-Fortran")
     variant('hdf4',    default=False, description="Enable HDF4 support")
 
     # Dependencies:
     depends_on("curl")  # required for DAP support
     depends_on("hdf", when='+hdf4')
-    depends_on("hdf5")  # required for NetCDF-4 support
+    depends_on("hdf5+mpi~cxx", when='+mpi')  # required for NetCDF-4 support
```

Comment:
```
 # mpi and cxx aren't compatible
```
that is certainly not the case. 

I think what that comment in `HDF5` is meant to say is that `c++` interface of `MPI` is not as complete as `c`, but nobody restricts the programmer to use `c++` interface in `c++` code.
You can call MPI's `c` function like `MPI_Allreduce` from a program compiled with `c++` with different standards like `c++11` or `c++14`. It is done all the time in the `deal.ii` library which is a `c++` library, see for example https://github.com/dealii/dealii/blob/master/include/deal.II/base/mpi.h#L524